src/literal.js

```diff
@@ -106,7 +93,8 @@ class Literal extends Node {
     } else {
       datatype = XSD.double
     }
-    return new Literal(strValue, null, datatype)
+    return Term.literalByValue(strValue, null, datatype) ||
+      Term.addLit(Term.literalByValue(strValue, null, datatype))
```

Comment:
`Term.literalByValue` already calls `Term.addLit`, so this call shouldn't be necessary